skyportal/plot.py

```diff
@@ -717,22 +716,25 @@ def spectroscopy_plot(obj_id, spec_id=None):
 
     color_map = dict(zip([s.id for s in spectra], viridis(len(spectra))))
 
-    data = pd.concat(
-        [
-            pd.DataFrame(
-                {
-                    'wavelength': s.wavelengths,
-                    'flux': s.fluxes,
-                    'id': s.id,
-                    'telescope': s.instrument.telescope.name,
-                    'instrument': s.instrument.name,
-                    'date_observed': s.observed_at.date().isoformat(),
-                    'pi': s.assignment.run.pi if s.assignment is not None else "",
-                }
-            )
-            for i, s in enumerate(spectra)
-        ]
-    )
+    data = []
+    for i, s in enumerate(spectra):
+
+        # normalize spectra to a common total flux (facilitates easy visual comparison)
+        normfac = np.sum(np.gradient(s.wavelengths) * s.fluxes)
```

Comment:
There is no guarantee that this is non-zero and finite. Please handle these cases.

Reply:
fixed